samcli/commands/deploy/guided_context.py

```diff
@@ -285,51 +288,211 @@ def prompt_parameters(
+
+        self.prompt_delete_unreferenced_repos(
+            [manager_helper.manager.get_repo_uri(repo) for repo in manager_helper.unreferenced_repos]
+        )
+
+        updated_repositories = manager_helper.remove_unreferenced_repos_from_mapping(updated_repositories)
+        GuidedContext.verify_images_exist_locally(self.function_provider.functions)
+
+        manager_helper.manager.sync_repos()
+        return updated_repositories
+
+    def prompt_specify_repos(
+        self,
+        functions_without_repos: List[str],
+        image_repositories: Dict[str, str],
+    ) -> Dict[str, str]:
+        """
+        Show prompts for each function that isn't associated with a image repo
+
+        Parameters
+        ----------
+        functions_without_repos: List[str]
+            List of functions without associating repos
+
+        image_repositories: Dict[str, str]
+            Current image repo dictionary with function logical ID as key and image repo URI as value.
+
+        Returns
+        -------
+        Dict[str, str]
+            Updated image repo dictionary with values(image repo URIs) filled by user input
+        """
+        updated_repositories = image_repositories.copy()
+        for function_logical_id in functions_without_repos:
+            image_uri = prompt(
+                f"\t {self.start_bold}ECR repository for {function_logical_id}{self.end_bold}",
+                type=click.STRING,
+            )
+            if not is_ecr_url(image_uri):
+                raise GuidedDeployFailedError(f"Invalid Image Repository ECR URI: {image_uri}")
+
+            updated_repositories[function_logical_id] = image_uri
+
+        return updated_repositories
+
+    def prompt_create_all_repos(self, functions: List[str], functions_without_repo: List[str]) -> bool:
+        """
+        Prompt whether to create all repos
+
+        Parameters
+        ----------
+        functions: List[str]
+            List of function logical IDs that are image based
+        functions_without_repo: List[str]
+            List of function logical IDs that do not have an ECR image repo specified
+
+        Returns
+        -------
+        Boolean
+            Returns False if there is no missing function or denied by prompt
+        """
+        if not functions:
+            return False
+
+        if functions == functions_without_repo:
+            click.echo("\t Image repositories: Not found.")
+            click.echo(
+                "\t #Managed repositories will be deleted when "
+                "their functions are removed from the template and deployed"
+            )
+            return confirm(
+                f"\t {self.start_bold}Create managed ECR repositories for all functions?{self.end_bold}", default=True
+            )
+
+        functions_with_repo_count = len(functions) - len(functions_without_repo)
+        click.echo(
+            "\t Image repositories: "
```

Comment:
Should we showcase the mapping of function to repos?

Reply:
Added